### ISSUE TITLE
Dota 2 crashes, then gets stuck on "Downloading Content 0%" when attempting to relaunch

### ISSUE THREAD

**Issue by FakeSalamander** [ValveSoftware/Dota-2#2839]
#### Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/): [My system information](https://gist.github.com/FakeSalamander/5a5414efedecdc52450feac60437e5e3)

* Have you checked for system updates?:  Yes
* Are you using the latest stable video driver available for your system? Yes
* Have you verified the game files?: Yes. verification hangs at 0% while the bug is ocurring, and finishes with no errors once it ends.

#### Please describe your issue in as much detail as possible:
Sometimes the game will randomly crash: either as I'm loading into a match, halfway through character select, or in the first few minutes of a match.   I think it always happen sometime after queuing up, but I'm not sure.

Once the game is closed, it will not reopen again. Attempting to click Play on steam just changes the text to "Downloading Content 0%", which it stays stuck on forever.  Eventually, in about 10-15 minutes, the issue mysteriously ends and I can reopen the game, but by then I've already gotten an unavoidable abandon.

I have tried restarting steam,  verifying integrity of game files, changing my download region, clearing the download cache, restarting my computer, and none of these have had any effect in ending the bug.

 Uninstalling the game while the bug is going  on is possible, but it then becomes impossible to download, stuck on an unresponsive  "Download" button. Other games can be downloaded and installed just fine while this is going on.  Once the 10-15 minutes are up, the game starts downloading as normal in this case.
 
 (Sidenote:: this bug is also occurring to another friend of mine, exactly as described above. His PC hardware is very different, with a NVIDIA GPU and a 12th-gen Intel CPU, but he also runs Arch Linux)

#### Steps for reproducing this issue:

1. Play Dota 2 until the game crashes.
2. Attempt to relaunch the game; notice that the game-launching sequence gets stuck on "Downloading content 0%"
3. Wait 10-15 minutes for the bug to end.

**Comment by massatt212** [comment]
i downgraded to kernel 6.6.58 longterm "lts" and dota works again, hopefully it fixes it for you also

**Comment by FakeSalamander** [comment]
were you having the same bug, or a different one?

**Comment by massatt212** [comment]
> were you having the same bug, or a different one?

Yes same problem, I just double check with latest linux-zen and game crashed and started to verify.

Testing 6.10.3 I'll get back to you.

**Comment by FakeSalamander** [comment]
Thank you. CachyOS is a bit odd about kernels, but it has a 6.6.57-3 lts kernel I can try. Will let you know how it goes.

I also found another thread involving the same bug, which I unfortunately missed earlier. I'd like to link to it here, but I don't know how.

**Comment by massatt212** [comment]
> Thank you. CachyOS is a bit odd about kernels, but it has a 6.6.57-3 lts kernel I can try. Will let you know how it goes.
> 
> I also found another thread involving the same bug, which I unfortunately missed earlier. I'd like to link to it here, but I don't know how.

I'm using cachyos also 😂

**Comment by massatt212** [comment]
6.10.3 works also if u don't want to go so low in kernel version.

**Comment by light-on-shadow** [comment]
Same here, Silverblue 6.11.6-300.fc41.x86_64, horrendous experience.

**Comment by FakeSalamander** [comment]
Switching to the cachyOS lts kernel seems to have fixed it for now,  I haven't had any crash since. 
Hoping this will get fixed for the newer kernel soon, but ATM it's just a relief to be able to play Dota again.

I don't know much about Fedora Silverblue, but you should try using Silverblue's LTS kernel and see if it helps too.

**Comment by light-on-shadow** [comment]
There are no distro specific kernels, it's one linux kernel with a nice packaging around it called a distro.
Which kernel are you running? (uname -r)

// nevermind saw that didn't see you were the OP, 6.11.x onwards seems to be quite buggy with AMD hardware, I've had a host of issues.

**Comment by light-on-shadow** [comment]
Well, hard to believe but seems like RAM was the issue in my case.
All of a sudden, yesterday, PC wouldn't even POST. 
Tried one stick then another and it came back to life, dota stopped crashing too.

There have been other updates but I had the same issue with Deadlock and it's gone too.
A bit weird only Valve games were the canary in the coalmine.

I'm running with one stick for now, I'll know for sure when the new RAM kit arrives.

**Comment by Kuechlin** [comment]
i had the same issue with kernel 6.11.7-200 for me the fix was to use [steam client beta](https://help.steampowered.com/en/faqs/view/276C-85A0-C531-AFA3)

**Comment by JoelMahon** [comment]
I had this issue on windows 10, after trying various things like restarting my PC, logging out of steam, clearing download cache, etc. nothing worked. I tried verifying game files, which took a couple more mins to stop being stuck at 0% and then eventually after a while "worked".

to me this matches what the original poster said, sucks extra hard if it was a time delay because I might have made it in time if I DIDN'T verify game files... sad

**Comment by MungTheWicked** [comment]
I have a lot to add regarding this issue. I've been dealing with it for years on multiple Linux distros: HoloISO, Ubuntu Studio LTS 24.0.2, and recently Ubuntu 24.10. I can't reliably play Ranked games as a result of this bug. I have a workaround for this bug where I switch to my MacBook Pro to play out the remainder of the match. Once the match is over the "Downloading Content 0%" on my Ubuntu comp will unfreeze and finish downloading the update, which is usually only 10's of Mb large. These 10's of Mb large files that get downloaded seem to be some kind of regular occurrence on Linux. I never observed them on any Windows comps I used in the past. I don't observe those same downloads on MacOS either.

Furthermore, if this bug occurs when you go to accept the match and you get booted out of the game, resulting in a red "Play Dota" button, the "Downloading Content 0%" will unfreeze the moment the timer associated with the red button reverting back to a green button occurs.

So, that "Wait 10-15 minutes for the bug to end" is either due to the match having ended at that time or the cooldown on the red "Play Dota" button having expired, allowing you to queue up again.

My current Linux specs are the following:
**Software**
KDE Plasma Version: 6.1.5
KDE Frameworks Version: 6.6.0
Qt Version: 6.6.2
Kernel Version: 6.11.0-1010-lowlatency (64-bit)

**Hardware**
Processors: 32 x AMD Ryzen 9 5950X 16-Core Processor
Memory: 64 Gb
Graphics Processor: NVIDIA GeForce RTX 3090 Ti/PCIe/SSE2

EDIT - 3/24/25
I sat through more than an hour of memtest86+ and got a "PASSED" result regarding my RAM.

**Comment by XiongDWM** [comment]
same probs, when it crashing it says vpk01.pak lost, but I got my network finely connected, in one work, while this game crashed , only thing i could do is uninstall the game and re install again...

**Comment by yuryfunikov** [comment]
as far as i can tell it's easier to fight game crashes issue instead of 'downloading content', analyze system logs (journalctl -b -1), in my case it showed problems with nvidia driver and mb bios so setting vidia-drm.modeset=1, updating bios and nvidia drivers seem to fixed the game crashing
